### PR TITLE
feat(cluster): impove error message when bootstrap channel fails

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

A cluster was reporting following error:
```
RaftServer{raft-partition-partition-3} - ConfigureRequest{...} to 0 failed: java.util.concurrent.CompletionException: java.net.ConnectException: Failed to connect channel for address zeebe-0.zeebe-broker-service.a7151101-b5c5-4415-8970-684b97779c02-zeebe.svc.cluster.local:26502
```
This has no information to help debug. This PR adds the resolved IP and error to this message.

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java [hunk 1]
```diff
@@ -725,7 +725,9 @@ private CompletableFuture<Channel> bootstrapClient(final Address address) {
                   if (!onConnect.isSuccess()) {
                     future.completeExceptionally(
                         new ConnectException(
-                            String.format("Failed to connect channel for address %s", address)));
+                            String.format(
+                                "Failed to connect channel for address %s (resolved: %s) : %s",
+                                address, address.address(), onConnect.cause())));
```
Comment (Zelldon) [inline atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java:730]: Lets follow our error guidelines :) https://github.com/camunda/zeebe/wiki/Error-Guidelines

```suggestion
                            String.format(
                                "Expected to create channel to %s (resolved: %s), but failed with : %s",
                                address, address.address(), onConnect.cause())));
```

**Review by ChrisKujawa — APPROVED** [review]
Thanks @deepthidevaki :)

File: atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java (hunk 1 above)
Comment (deepthidevaki) [reply atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java:730]: I find it unnecessarily verbose. When we say "Failed to connect channel" It is already clear that it is expected to connect channel. Right? Besides the error guidelines doesn't require to use the exact same format.

Comment (Zelldon) [reply atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java:730]: I prefer a consistent format, even if it is a bit more verbose it is always clear what is expected and what happened with additional context. As far as I know, we agreed on that a while ago. If you disagree, feel free to bring it up in a team discussion.

Comment (Zelldon) [reply atomix/cluster/src/main/java/io/atomix/cluster/messaging/impl/NettyMessagingService.java:730]: BTW go ahead to merge it anyway, nothing what should block you (I already accepted)

**Comment by deepthidevaki** [comment]
bors merge

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**